### PR TITLE
Wait pipeline image ready before build src

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-826

/cc @openshift/openshift-team-developer-productivity-test-platform

### CODE REVIEW

File: pkg/steps/source.go [hunk 1]
```diff
@@ -257,6 +258,18 @@ func buildFromSource(jobSpec *api.JobSpec, fromTag, toTag api.PipelineImageStrea
...
+	log.Printf("waiting for importing %s ...", from.Name)
+	if err := wait.Poll(10*time.Second, 5*time.Minute, func() (bool, error) {
+		pipeline, err := imageClient.ImageStreams(jobSpec.Namespace).Get(api.PipelineImageStream, meta.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		_, exists := util.ResolvePullSpec(pipeline, string(fromTag), true)
+		return exists, nil
+	}); err != nil {
+		log.Printf("could not resolve imagestream %s: %v", from.Name, err)
```
Comment (hongkailiu) [inline pkg/steps/source.go]: @stevekuznetsov 

Should it `panic()` here?

**Comment by hongkailiu** [comment]
/hold

let me rehearse with the built image.

**Comment by hongkailiu** [comment]
Tested the code by https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/7224/rehearse-7224-pull-ci-integr8ly-cloud-resource-operator-master-vendor/7#1:build-log.txt%3A14

**Comment by bbguimaraes** [comment]
Feel free to ignore for now if this is blocking more important work, but can we:
- use a watch instead of polling?
- poll/watch the specific `ImageStreamTag` instead of the entire `ImageStream`?
- use/refactor the `Done` method (which is currently dead code, as already mentioned elsewhere)?
- add a test?

**Comment by petr-muller** [comment]
> Feel free to ignore for now if this is blocking more important work, but can we:

I'd prefer these to not be ignored. We have enough tech debt in ci-operator already and should really not create more, IMO. The items you posted are all relevant.

**Comment by hongkailiu** [comment]
> * use a watch instead of polling?
> * poll/watch the specific `ImageStreamTag` instead of the entire `ImageStream`?

Asked dev-ex: `isTag` create is not a synced call. We need to check the `status.tags` of the `is` (not `isTag` but i did not ask why) as we do in `util.ResolvePullSpec`. What we do in `ResolvePullSpec` is existing code. I hope i do not introduce more debts by reusing it.

Watch could be working too (also asked dev-ex). But they use `wait.pull` [in their code](https://github.com/openshift/origin/blob/master/test/extended/util/framework.go#L294). 

> * use/refactor the `Done` method (which is currently dead code, as already mentioned elsewhere)?

I do not have answer for this. What is the plan for `Done` method? Do we want to remove it or use it more properly?

I could implement import-checking in the `Done` method and call it in `run`. However, it does not help if the plan is to remove it.

> * add a test?

Fixing the existing test already is a test case for the changed code.
Do you want me to add more tests? The one that I can think of

* the status is not ready until for example 30 secs after the isTag is created

Let me know if you want me to add the above case, or other cases that I missed.

**Comment by bbguimaraes** [comment]
I'm fine with polling and the test. As I said, those changes are not strictly necessary, and migrating to the build server is orders of magnitude more important at the moment.

If we're not able to use `Done` unchanged, then I see no reason to mess with it.

**Pull request merged**